### PR TITLE
Include tinyformat as a subtree

### PR DESCRIPTION
**Pull request by Empact**
This adds the existing c42f/tinyformat@689695c of tinyformat as a subtree, and extracts the modifications that had been made to a utilstrprintf.h wrapper file which is included instead of tinyformat.h directly.

The benefits being that this should be easier to verifiably update and clearer what changes are being introduced, while making it easier to identify tinyformat.h as a dependency rather than a project file, and apply special treatment relative to that fact.

### CODE REVIEW

File: src/Makefile.am [hunk 1]
```diff
@@ -436,8 +438,8 @@ nodist_libbitcoin_util_a_SOURCES = $(srcdir)/obj/build.h
 #
 
 # bitcoind binary #
-bitcoind_SOURCES = bitcoind.cpp
-bitcoind_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES)
+bitcoind_SOURCES = bitcoind.cpp tinyformat/tinyformat.h
```
Comment (Empact) [inline src/Makefile.am]: I'm unclear on why this is required to be added to sources, but the build failed consistently when it was only provided via `-I` below, e.g.:
https://travis-ci.org/Empact/bitcoin/jobs/410971896

**Comment by Empact** [comment]
Re-opened from #13834. This is a step toward creating an separate compilation context for bitcoin vs dependency code, which will allow for stricter internal checks during compilation.

**Comment by MarcoFalke** [comment]
Seems like a lot of files to pull in just to make it a subtree, no? Imo tinyformat is really meant to be a single file just to be copied into the project.

**Comment by Empact** [comment]
@MarcoFalke I dropped the undesired files in 993d88104d73c985b29d9356e3347d92f76b6a61, now it's just the `.h` and readme.

**Comment by sipa** [comment]
What's the point of using a subtree if you're not actually going to include the exact tree?

**Comment by Empact** [comment]
This now includes the full, checked subtree, and I dropped a few other incidental commits in hopes of reducing the conflicts noted above.

**Comment by Empact** [comment]
Reopening as a new PR so as to calculate new conflicts.

**Pull request closed without merging**